doc/audit/security_auditing_HLD.md

```diff
@@ -0,0 +1,513 @@
+    - [3.1 Design](#31-design)
+    - [3.2 Audit Rules Review](#32-audit-rules-review)
+    - [3.3 Configuration design](#33-configuration-design)
+      - [3.3.1 ConfigDB schema](#331-configdb-schema)
+        - [3.3.1.1 AUDIT TABLE](#3311-audit-table)
+        - [3.3.1.2 Config DB JSON Sample](#3312-config-db-json-sample)
+        - [3.3.1.3 Redis Entries Sample](#3313-redis-entries-sample)
+      - [3.3.2 YANG model](#332-yang-model)
+      - [3.3.3 Flows](#333-flows)
+        - [3.3.3.1 Audit Init Flow](#3331-audit-init-flow)
+        - [3.3.3.2 Audit Config Flow](#3332-audit-config-flow)
+        - [3.3.3.3 Audit Show Flow](#3333-audit-show-flow)
+      - [3.3.4 CLI design](#334-cli-design)
+      - [3.3.5 Logrotate](#335-logrotate)
+    - [3.6 Warmboot and Fastboot Design Impact](#36-warmboot-and-fastboot-design-impact)
+    - [3.7 Timeline](#37-timeline)
+    - [3.8 Performance](#38-performance)
+    - [3.9 Audit Rule Order](#39-audit-rule-order)
+    - [3.10 Security Compliance](#310-security-compliance)
+  - [4. Testing Requirements/Design](#4-testing-requirementsdesign)
+    - [4.1 Unit Test cases](#41-unit-test-cases)
+    - [4.2 System Test cases](#42-system-test-cases)
+  - [5. Q\&A](#5-qa)
+  - [6. Future Improvements](#6-future-improvements)
+
+## List of Tables
+* [Table 1: Revision](#table-1-revision)
+* [Table 2: Audit Rules Review](#table-2-audit-rules-review)
+* [Table 3: Unit Test Cases](#table-3-unit-test-cases)
+* [Table 4: System Test Cases](#table-4-system-test-cases)
+
+## Revision
+###### Table 1: Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 08/20/2024  | Mai Bui            | Initial version                   |
+
+## Scope
+This section describes the audit enhancement high-level design in SONiC.
+
+## 1. Overview
+This design aims to enhance the auditing capabilities within SONiC operating system using audit daemon (auditd). Auditing is the process of recording and analyzing the events that occur on the device. Auditing can help to detect unauthorized access, configuration changes, malicious activity, or system errors. Auditing can also provide evidence for forensic investigations, compliance audits, or incident response.
+
+This document outlines the addition of several auditd rules to enhance the auditing capabilities on a SONiC device. These extra rules aim to provide thorough monitoring of important system activities, making the device more secure and better prepared for compliance and incident response.
+
+## 2. Current Architecture Design
+In SONiC, audit settings are centrally managed through a configuration file at `/etc/audit/auditd.conf`. Upon system startup, SONiC automatically compiles audit rules from the `/etc/audit/rules.d/` directory into a single file, `/etc/audit/audit.rules`, which auditd then loads. Here is a brief look at the relevant files:
```

Comment:
Would be better if have a diagram for current architecture?

Reply:
Good idea, but actually, I added proposed architecture diagram, I think this way is better